barretenberg/cpp/src/barretenberg/commitment_schemes/gemini/gemini.hpp

```diff
@@ -234,21 +241,36 @@ template <typename Curve> class GeminiProver_ {
                 for (size_t i = 0; i < groups_to_be_interleaved[0].size(); ++i) {
                     batched_group.push_back(Polynomial(full_batched_size));
                 }
+
+                // Pre-compute all scalars
+                std::vector<Fr> scalars;
+                scalars.reserve(groups_to_be_interleaved.size());
+                Fr current_scalar = running_scalar;
                 for (size_t i = 0; i < groups_to_be_interleaved.size(); ++i) {
-                    batched_interleaved.add_scaled(interleaved[i], running_scalar);
-                    for (size_t j = 0; j < groups_to_be_interleaved[0].size(); ++j) {
-                        batched_group[j].add_scaled(groups_to_be_interleaved[i][j], running_scalar);
-                    }
-                    running_scalar *= challenge;
+                    scalars.push_back(current_scalar);
+                    current_scalar *= challenge;
+                }
+                running_scalar = current_scalar;
+
+                for (size_t i = 0; i < groups_to_be_interleaved.size(); ++i) {
+                    batched_interleaved.add_scaled(interleaved[i], scalars[i]);
```

Comment:
Why does this need precomputation of scalars? 
it looks like we could have kept running_scalar in this paradigm?